### PR TITLE
raft: port disruptive test

### PR DESCRIPTION
**Pull request by july2993**
the port the raft relate part in etcd about https://github.com/pingcap/raft-rs/issues/39

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: tests/cases/test_raft.rs [hunk 1]
```diff
@@ -2260,6 +2260,154 @@ fn test_non_promotable_voter_which_check_quorum() {
     assert_eq!(nt.peers[&2].leader_id, 1);
 }
 
+#[test]
+// test_disruptive_follower tests isolated follower,
```
Comment (BusyJay) [inline tests/cases/test_raft.rs]: Please use [doc comment](https://doc.rust-lang.org/reference/comments.html#doc-comments) and put the `#[test]` attribute below the comments.

File: tests/cases/test_raft.rs [hunk 2]
```diff
@@ -2260,6 +2260,154 @@ fn test_non_promotable_voter_which_check_quorum() {
...
+    assert_eq!(nt.peers[&1].state, StateRole::Leader);
+    assert_eq!(nt.peers[&2].state, StateRole::Follower);
+    assert_eq!(nt.peers[&3].state, StateRole::Follower);
+
+    nt.isolate(3);
+    nt.send(vec![new_message(1, 1, MessageType::MsgPropose, 1)]);
+    nt.send(vec![new_message(1, 1, MessageType::MsgPropose, 1)]);
+    nt.send(vec![new_message(1, 1, MessageType::MsgPropose, 1)]);
+
+    nt.peers.get_mut(&1).unwrap().pre_vote = true;
```
Comment (BusyJay) [inline tests/cases/test_raft.rs]: I think this can be set when the node is created using `new_test_raft_with_prevote`.

File: src/raft.rs [hunk 3]
```diff
@@ -224,7 +224,7 @@ pub struct Raft<T: Storage> {
     heartbeat_elapsed: usize,
 
     pub check_quorum: bool,
-    pre_vote: bool,
+    pub pre_vote: bool,
```
Comment (BusyJay) [inline src/raft.rs]: Seems it doesn't have to be `pub` now?

**Comment by BusyJay** [comment]
Thanks for your contributions!

**Pull request merged**